g3doc/Rationale.md

```diff
@@ -794,14 +794,13 @@ At a high level, we have two alternatives here:
     explicitly propagate the schedule into domains and model all the cleanup
     code. An example and more detail on the schedule tree form is in the next
     section.
-1.  Having two different forms of MLFunctions: an affine loop tree form
+1.  Having two different forms of "affine regions": an affine loop tree form
     (AffineLoopTreeFunction) and a polyhedral schedule tree form as two
-    different forms of MLFunctions. Or in effect, having four different forms
-    for functions in MLIR instead of three: CFG Function,
-    AffineLoopTreeFunction, Polyhedral Schedule Tree function, and external
-    functions.
+    different forms. Or in effect, having four different forms for functions in
+    MLIR instead of three: CFG Function, AffineLoopTreeFunction, Polyhedral
+    Schedule Tree function, and external functions.
```

Comment:
The paragraph is really documenting history on the design choices explored. I don't think we need to now update it.

Reply:
I don't disagree with keeping some history, but the text as-is does not seem to be understandable by someone not already knowing the history. For example this is the only reference in the whole document to "CFG Function". 

I think the whole "rationale" document should either be geared towards explaining the rational for what is currently in the repo, and understandable for someone reading the existing code (with very limited historical anecdotes), or alternatively trying to provide an "historical evolutionary" view of the codebase.

If we don't review this document and make it so that it can be maintained and evolve with the rest of the documentation, I'm not even sure we should keep it at all.

Reply:
Yes, I agree with you. This document does require an update. I didn't intend to do that in this PR, and I shouldn't have probably touched this part of the file (I just wanted to fix those memory space names above).